CardScan/Classes/VideoFeed.swift

```diff
@@ -176,10 +176,24 @@ class VideoFeed {
         videoDevice.unlockForConfiguration()
     }
     
+    //MARK: -Torch Logic
     func toggleTorch() {
         self.torch?.toggle()
     }
     
+    func isTorchOn() -> Bool {
+        return self.torch?.state ==  Torch.State.on ? true : false
+    }
+    
+    func hasTorch() -> Bool {
+        return self.torch?.device?.hasTorch ?? false
```

Comment:
I don't know if `hasTorch` or `hasTorchAndIsAvailable` is better, do you have an opinion?

Reply:
hmmmm... i guess hasTorch&isAvailable is the safer route. So I'll go with that.

Reply:
👍